src/platform/packages/private/kbn-grid-layout/grid/use_grid_layout_events/panel/utils.ts

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+import type { GridLayoutStateManager, PanelInteractionEvent } from '../../types';
+import type { UserInteractionEvent, PointerPosition } from '../types';
+import { KeyboardCode, type UserKeyboardEvent } from '../sensors/keyboard/types';
+import { getSensorPosition, isKeyboardEvent, isMouseEvent, isTouchEvent } from '../sensors';
+
+// Calculates the preview rect coordinates for a resized panel
+export const getResizePreviewRect = ({
+  interactionEvent,
+  pointerPixel,
+  maxRight,
+}: {
+  pointerPixel: PointerPosition;
+  interactionEvent: PanelInteractionEvent;
+  maxRight: number;
+}) => {
+  const panelRect = interactionEvent.panelDiv.getBoundingClientRect();
+
+  return {
+    left: panelRect.left,
+    top: panelRect.top,
+    bottom: pointerPixel.clientY - interactionEvent.sensorOffsets.bottom,
+    right: Math.min(pointerPixel.clientX - interactionEvent.sensorOffsets.right, maxRight),
+  };
+};
+
+// Calculates the preview rect coordinates for a dragged panel
+export const getDragPreviewRect = ({
+  pointerPixel,
+  interactionEvent,
+}: {
+  pointerPixel: PointerPosition;
+  interactionEvent: PanelInteractionEvent;
+}) => {
+  return {
+    left: pointerPixel.clientX - interactionEvent.sensorOffsets.left,
+    top: pointerPixel.clientY - interactionEvent.sensorOffsets.top,
+    bottom: pointerPixel.clientY - interactionEvent.sensorOffsets.bottom,
+    right: pointerPixel.clientX - interactionEvent.sensorOffsets.right,
+  };
+};
+
+// Calculates the sensor's offset relative to the active panel's edges (top, left, right, bottom).
+// This ensures the dragged or resized panel maintains its position under the cursor during the interaction.
+export function getSensorOffsets(e: UserInteractionEvent, { top, left, right, bottom }: DOMRect) {
+  if (!isTouchEvent(e) && !isMouseEvent(e) && !isKeyboardEvent(e)) {
+    throw new Error('Unsupported event type: only mouse, touch, or keyboard events are handled.');
+  }
+  const { clientX, clientY } = getSensorPosition(e);
+  return {
+    top: clientY - top,
+    left: clientX - left,
+    right: clientX - right,
+    bottom: clientY - bottom,
+  };
+}
+
+const KEYBOARD_DRAG_BOTTOM_LIMIT = 8; // The offset from the bottom of the page to ensure the dragged handle doesn't look 'glued' to the page
```

Comment:
Do we want to use an EUI size variable here?

Reply:
Actually I found out that this is not needed with `scroll-margin-bottom` css we use so I removed it :)